### ISSUE TITLE
Issue with insertion with DEFAULT value with H2 and onDuplicateKeyIgnore

### ISSUE THREAD

**Issue by blacelle** [jOOQ/jOOQ#7648]
### Steps to reproduce the problem (include example code if possible):

I insert a table with :

    CREATE TABLE IF NOT EXISTS table_name (id UUID, data TEXT,timestamp TIMESTAMP WITH TIME ZONE DEFAULT NOW(),PRIMARY KEY (id))

Then I try to insert with a default value for timestamp:

    create
					.insertInto(TableName.TABLE_NAME,
							TableName.TABLE_NAME.id,
							TableName.TABLE_NAME.data,
							TableName.TABLE_NAME.timestamp)
					.values(DSL.val(someId),
							DSL.val(someData),
							DSL.defaultValue(
									TableName.TABLE_NAME.TIMESTAMP))
					.onDuplicateKeyIgnore()
					.execute();

It leads to the stack:

        org.jooq.exception.DataAccessException: SQL [insert into "table_name" ("id", "shared_data", "timestamp") select "t"."id", "t"."data", "t"."timestamp" from (select cast(? as uuid) "id", cast(? as varchar) "data", default "timestamp" where not exists (select 1 "one" from "table_name" where "table_name"."id" = cast(? as uuid))) "t"]; Column "default" not found; SQL statement:
    CREATE FORCE VIEW PUBLIC._0 AS
    SELECT
        CAST(?1 AS UUID) AS "id",
        CAST(?2 AS VARCHAR) AS "data",
        default AS "timestamp"
    FROM SYSTEM_RANGE(1, 1)
    WHERE NOT EXISTS(
        SELECT
            1 AS "one"
        FROM PUBLIC."table_name"
        WHERE table_name."id" = CAST(?3 AS UUID)) [42122-197]
        at org.jooq_3.10.7.H2.debug(Unknown Source)
        at org.jooq.impl.Tools.translate(Tools.java:2241)
        at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:704)
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:361)
        at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:127)

### Versions:

- jOOQ:3.10.7
- Java: 1.8
- Database (include vendor): H2 1.4.197
- OS: MacOS
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thank you very much for your report. This is a curious case of using the `DEFAULT` expression, which breaks our `ON DUPLICATE KEY IGNORE` emulation using `INSERT .. SELECT`. I guess in the event of emulating the clause, the columns that have `DEFAULT` expressions must simply not be inserted into. The question remains, what happens in the case of a multi row insert that contains `DEFAULT` expressions?

Another way to emulate the `DEFAULT` expression in places where it is syntactically invalid would be to run a subquery like:

```sql
(
  SELECT column_default 
  FROM information_schema.columns 
  WHERE schema_name = '...'
  AND table_name = '...' 
  AND column_name = '...'
)
```

**Comment by lukaseder** [comment]
The multi row insert statement is really the problem here when emulating `INSERT .. ON DUPLICATE KEY IGNORE` using `INSERT .. SELECT`.

More recent versions of H2 support `MERGE` (which we'll generate here, from now on: https://github.com/jOOQ/jOOQ/issues/10460), but the `MERGE` emulation has the same problem.

**Comment by lukaseder** [comment]
This seems to be no longer a problem in H2's `MERGE` implementation of `ON DUPLICATE KEY IGNORE`, which is available starting from jOOQ 3.14. This query:

```java
create().insertInto(TAuthor(), TAuthor_ID(), TAuthor_FIRST_NAME(), TAuthor_LAST_NAME())
        .values(val(3), DSL.default_(TAuthor_FIRST_NAME()), val("Koontz"))
        .onDuplicateKeyIgnore()
        .execute();
```

Produces this valid H2 SQL:

```sql
merge into "PUBLIC"."T_AUTHOR"
using (
  select 1 "one"
)
on (
  "PUBLIC"."T_AUTHOR"."ID" = 3
  or "PUBLIC"."T_AUTHOR"."LAST_NAME" = 'Koontz'
)
when not matched then insert (
  "ID",
  "FIRST_NAME",
  "LAST_NAME"
)
values (
  3, 
  default, 
  'Koontz'
)
```

I'm closing this for now. Other dialects may still have the issue, but I think the simplest solution here would be to just avoid setting the `default` value in the first place.
